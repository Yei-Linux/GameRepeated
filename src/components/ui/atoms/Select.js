import classNames from 'classnames'

const Select = ({
  ariaLabel,
  variation = 'primary',
  options,
  name,
  defaultValue,
  onChange,
  padding = 1,
  placeholder,
  css,
  className,
}) => {
  return (
    <select
      aria-label={ariaLabel}
      onChange={onChange}
      placeholder={placeholder}
      style={css}
      className={classNames(
        className,
        'select',
        `select--pd${padding}`,
        `select--${variation}`
      )}
      name={name}
      value={defaultValue}
    >
      {options.map(({ value, label, testId }) => (
        <option data-testid={testId} key={value} value={value}>
          {label}
        </option>
      ))}
    </select>
  )
}

export default Select
