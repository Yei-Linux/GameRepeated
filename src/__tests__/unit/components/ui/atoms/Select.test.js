import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Select from '../../../../../components/ui/atoms/Select'

describe('The Select Component', () => {
  const defaultProps = {
    ariaLabel: 'selectTest',
    variation: 'primary',
    options: [
      { value: 'optionOne', label: 'optionOne', testId: 'optionOne' },
      { value: 'optionTwo', label: 'optionTwo', testId: 'optionTwo' },
    ],
    name: 'selectTest',
    defaultValue: '',
    padding: 1,
    placeholder: 'input',
    css: {},
    className: 'inputCustom',
    onChange: jest.fn(),
  }

  const setupComponent = (props = defaultProps) => render(<Select {...props} />)

  it('Should reder input component', () => {
    const { asFragment } = setupComponent()

    expect(asFragment()).toMatchSnapshot()
  })
  it('Should reder input component', async () => {
    const mockOnChange = jest.fn(({ target: { value } }) => value)
    setupComponent({ ...defaultProps, onChange: mockOnChange })

    const select = screen.getByRole('combobox', { name: 'selectTest' })
    const optionSelected = screen.getByRole('option', { name: 'optionTwo' })
    userEvent.selectOptions(select, optionSelected)

    expect(mockOnChange).toBeCalledTimes(1)
    expect(mockOnChange.mock.results[0].value).toStrictEqual('optionTwo')
  })
})
