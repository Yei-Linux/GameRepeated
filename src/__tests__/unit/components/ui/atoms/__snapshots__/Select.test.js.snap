// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Select Component Should reder input component 1`] = `
<DocumentFragment>
  <select
    aria-label="selectTest"
    class="inputCustom select select--pd1 select--primary"
    name="selectTest"
    placeholder="input"
  >
    <option
      data-testid="optionOne"
      value="optionOne"
    >
      optionOne
    </option>
    <option
      data-testid="optionTwo"
      value="optionTwo"
    >
      optionTwo
    </option>
  </select>
</DocumentFragment>
`;
